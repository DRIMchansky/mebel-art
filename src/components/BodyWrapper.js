// packages
import React from 'react'
import styled from 'styled-components'
// components
import Hamburger from '../components/Hamburger'
import Logo from '../components/Logo'
// utils
import { breakpoints, colors, shadows } from '../util/cssConfig'

const BodyWrapper = ({ children, active, setActive }) => {
  return (
    <BodyWrapperStyled active={active}>
      <MobileHeader>
        <Hamburger active={active} setActive={setActive} />
        <Logo />
      </MobileHeader>
      {children}
    </BodyWrapperStyled>
  )
}
const BodyWrapperStyled = styled.div`
  flex-grow: 1;
  height: 100%;
  display: flex;
  flex-direction: column;
  background: ${colors.white};
  z-index: 2;
  overflow-x: hidden;
  transform: ${(props) => (props.active ? 'translate(60%, 3%)' : 'none')};
  border-radius: ${(props) => (props.active ? '7px' : 'none')};
  transition: all 320ms;
  box-shadow: ${shadows.four};

  @media (min-width: ${breakpoints.medium}) {
  }
`

const MobileHeader = styled.header`
  position: relative;
  padding: 0.9rem 1.2rem;
  text-align: center;

  @media (min-width: ${breakpoints.medium}) {
    display: none;
    visibility: none;
  }
`

export default BodyWrapper
