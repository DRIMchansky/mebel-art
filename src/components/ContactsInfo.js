// packages
import React from 'react'
import styled from 'styled-components'
// icons
import PlaceIcon from '@material-ui/icons/Place'
import MailOutlineIcon from '@material-ui/icons/MailOutline'
import PhoneIphoneIcon from '@material-ui/icons/PhoneIphone'
// utils
import { colors } from '../util/cssConfig'

const ContactsInfo = () => {
  return (
    <InfoStyled>
      <ul>
        <li>
          <PlaceIcon />
          <span>Где мы работаем</span>
          <small>Ярославль и Ярославская область</small>
        </li>
        <li>
          <MailOutlineIcon />
          <span>example@gmail.com</span>
        </li>
        <li>
          <PhoneIphoneIcon />
          <span>(01) 666 - 693 - 456</span>
        </li>
      </ul>
    </InfoStyled>
  )
}

const InfoStyled = styled.div`
  color: ${colors.black};
  padding: 2rem;
  padding-left: 3rem;

  @media (min-width: 960px) {
    padding: 0;
    height: 100%;
    width: 100%;
    display: flex;
    justify-content: center;
    align-items: flex-start;
  }

  & ul {
    list-style-type: none;
    padding: 0;

    @media (min-width: 960px) {
      margin-top: 5rem;
    }
  }

  & li {
    position: relative;
    margin-top: 1rem;

    @media (min-width: 960px) {
      margin-top: 2rem;
    }
  }

  & span {
    font-weight: 400;
    display: block;
    font-size: 1.1rem;
    margin-bottom: 0.3rem;

    @media (min-width: 960px) {
      margin-bottom: 1rem;
    }
  }

  & small {
    font-weight: 300;
    font-size: 1rem;
    opacity: 0.8;
  }

  & svg {
    position: absolute;
    left: -2rem;
    top: -0.2rem;
    fill: ${colors.accent};
  }
`

export default ContactsInfo
