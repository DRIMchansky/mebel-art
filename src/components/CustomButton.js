// packages
import React from 'react'
import styled from 'styled-components'
// components
import { Button } from 'gatsby-theme-material-ui'

const CustomButton = ({ children, size, variant, bgcolor, textcolor, href }) => {
  return (
    <ButtonStyled variant={variant} size={size} bgcolor={bgcolor} to={href} textcolor={textcolor}>
      {children}
    </ButtonStyled>
  )
}

const ButtonStyled = styled(Button)`
  background: ${(props) => props.bgcolor} !important;
  color: ${(props) => props.textcolor} !important;
`

export default CustomButton
