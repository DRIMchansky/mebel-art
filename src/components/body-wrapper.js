// packages
import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'
// components
import Hamburger from './hamburger'
import Logo from './logo'
// utils
import { breakpoints, colors, shadows } from '../util/css-config'

const BodyWrapper = ({ children, active, setActive }) => {

  return (
    <SBodyWrapper active={active}>

      <MobileHeader>
        <Hamburger active={active} setActive={setActive} />
        <Logo />
      </MobileHeader>

      {children}

    </SBodyWrapper>
  )
}

BodyWrapper.propTypes = {
  children: PropTypes.oneOfType([PropTypes.array, PropTypes.element]),
  active: PropTypes.bool,
  setActive: PropTypes.func
}

const SBodyWrapper = styled.div`
  flex-grow: 1;
  height: 100%;
  display: flex;
  flex-direction: column;
  background: ${colors.white};
  z-index: 2;
  overflow: hidden;
  transform: ${(props) => (props.active ? 'translate(60%, 3%)' : 'none')};
  border-radius: ${(props) => (props.active ? '7px' : 'none')};
  transition: all 320ms;
  box-shadow: ${shadows.four};

  @media (min-width: ${breakpoints.medium}) {
  }
`

const MobileHeader = styled.header`
  position: relative;
  padding: 0.9rem 1.2rem;
  text-align: center;

  @media (min-width: ${breakpoints.small}) {
    display: none;
    visibility: hidden;
  }
`

export default BodyWrapper
