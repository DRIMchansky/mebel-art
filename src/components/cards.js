// packages
import React from 'react'
import { graphql, useStaticQuery } from 'gatsby'
import Img from 'gatsby-image'
import styled from 'styled-components'
// components
import Container from '@material-ui/core/Container'
import Grid from '@material-ui/core/Grid'
// utils
import { colors, shadows } from '../util/css-config'

const Cards = () => {
  const cardsData = useStaticQuery(graphql`
    query {
      allContentfulFeature {
        edges {
          node {
            id
            image {
              fluid {
                ...GatsbyContentfulFluid
              }
            }
            title
            description
          }
        }
      }
    }
  `).allContentfulFeature.edges

  return (
    <CardsStyled>
      <Container maxWidth="lg">
        {/* <Title>Почему с нами выгодно?</Title> */}
        <Grid container spacing={3} justify="space-around">
          {cardsData.map((item) => {
            return (
              <GridItemStyled key={item.node.title} item sm={6} md={4}>
                <CardStyled>
                  <ImageStyled fluid={item.node.image.fluid} />
                  <DescriptionStyled>
                    <h3>{item.node.title}</h3>
                    <p>{item.node.description}</p>
                  </DescriptionStyled>
                </CardStyled>
              </GridItemStyled>
            )
          })}
        </Grid>
      </Container>
    </CardsStyled>
  )
}

const CardsStyled = styled.section`
  padding: 3rem 0 4rem;
`

const GridItemStyled = styled(Grid)`
  max-width: 400px;
  flex-grow: 1;
  display: flex;
  flex-direction: column;
`

const CardStyled = styled.div`
  box-shadow: ${shadows.two};
  flex-grow: 1;
  display: flex;
  flex-direction: column;

  & h3 {
    font-size: 1.1rem;
    color: ${colors.dark};
    margin: 0.5rem 0;
  }
  & p {
    font-size: 0.8rem;
    margin: 0.5rem 0;
    color: ${colors.gray};
  }
`
const DescriptionStyled = styled.div`
  background: ${colors.white};
  padding: 0.5rem;
  flex-grow: 1;
  border-radius: 0 0 5px 5px;
`

const ImageStyled = styled(Img)`
  border-radius: 5px 5px 0 0;
  border-bottom: 4px solid ${colors.dark};
`

export default Cards
