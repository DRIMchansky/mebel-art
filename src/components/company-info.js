// packages
import React from 'react'
import styled from 'styled-components'
import { graphql, useStaticQuery } from 'gatsby'
// components
import Typography from './typography'
// utils
import { colors } from '../util/css-config'
import contentfulRichTextTransform from '../util/contentful-rich-text-transform'

const CompanyInfo = () => {
  const data = useStaticQuery(graphql`
    query {
      allContentfulGeneralInfo {
        edges {
          node {
            about_company {
              raw
            }
          }
        }
      }
    }
  `).allContentfulGeneralInfo.edges[0].node.about_company.raw

  return (
    <CompanyInfoStyled>
      <Typography html={contentfulRichTextTransform(data)} />
    </CompanyInfoStyled>
  )
}

const CompanyInfoStyled = styled.section`
  background: ${colors.white};
  border-radius: 5px;
  padding: 1rem 2rem;
`

export default CompanyInfo
