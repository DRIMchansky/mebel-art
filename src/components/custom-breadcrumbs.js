import React from 'react'
import PropTypes from 'prop-types'
import Typography from '@material-ui/core/Typography'
import Breadcrumbs from '@material-ui/core/Breadcrumbs'
import Link from '@material-ui/core/Link'

const CustomBreadcrumbs = ({ currentPage, previousPage, previousUrl }) => {

  return (
    <Breadcrumbs aria-label="breadcrumb" style={{ marginTop: 20 }}>
      <Link color="inherit" href="/">
        МЕБЕЛЬ АРТ
      </Link>
      {previousPage && <Link color="inherit" href={previousUrl}>{previousPage}</Link>}
      <Typography color="textPrimary">{currentPage}</Typography>
    </Breadcrumbs>
  )
}

CustomBreadcrumbs.propTypes = {
  currentPage: PropTypes.string,
  previousPage: PropTypes.string,
  previousUrl: PropTypes.string,
}

export default CustomBreadcrumbs
