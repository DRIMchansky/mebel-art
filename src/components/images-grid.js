// packages
import React from 'react'
import { graphql, useStaticQuery } from 'gatsby'
import styled from 'styled-components'
// components
import Container from '@material-ui/core/Container'
import Gallery from '@browniebroke/gatsby-image-gallery'
import Title from '../components/title'
import CustomBreadcrumbs from './custom-breadcrumbs'
// utilities
import '@browniebroke/gatsby-image-gallery/dist/style.css'

export const PureImagesGrid = ({ images }) => {
  return (
    <ImagesGridStyled>
      <Container maxWidth="md">
        <CustomBreadcrumbs currentPage="Наши проекты" />
        <Title>
          <h1>Фотографии наших работ</h1>
          <span>
            Вы можете увидеть больше или загрузить свои в нашей группе{' '}
            <a href="https://vk.com/mebelart76" target="_blank" rel="noreferrer">
              ВКонтакте
            </a>
          </span>
        </Title>
        <Gallery images={images} />
      </Container>
    </ImagesGridStyled>
  )
}

export const ImagesGrid = () => {
  const images = useStaticQuery(graphql`
    query {
      allContentfulGallery {
        edges {
          node {
            images {
              id
              thumb: fluid(maxWidth: 270, maxHeight: 270) {
                ...GatsbyContentfulFluid
              }
              full: fluid(maxWidth: 1024) {
                ...GatsbyContentfulFluid
              }
            }
          }
        }
      }
    }
  `).allContentfulGallery.edges[0].node.images

  return <PureImagesGrid images={images} />
}

const ImagesGridStyled = styled.section`
  padding-bottom: 2rem;

  & img {
    cursor: pointer;
  }
`

export default ImagesGrid
