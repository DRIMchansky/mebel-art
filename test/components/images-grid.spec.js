import React from 'react'
import renderer from 'react-test-renderer'
import { PureImagesGrid as ImagesGrid } from '../../src/components/images-grid'
import { StaticQuery } from 'gatsby'
import { shallow } from 'enzyme'

describe('ImagesGrid', () =>
  it('renders correctly', () => {
    const data = StaticQuery.mockImplementationOnce(({ render }) =>
      render({
        x: 'x',
      })
    )

    const tree = shallow(<ImagesGrid data={data} />)
    expect(tree).toMatchSnapshot()
  }))
