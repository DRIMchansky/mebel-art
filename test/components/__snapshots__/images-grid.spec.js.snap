// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImagesGrid renders correctly 1`] = `
<styled.section>
  <WithStyles(ForwardRef(Container))
    maxWidth="md"
  >
    <CustomBreadcrumbs
      currentPage="Наши проекты"
    />
    <Title>
      <h1>
        Фотографии наших работ
      </h1>
      <span>
        Вы можете увидеть больше или загрузить свои в нашей группе
         
        <a
          href="https://vk.com/mebelart76"
          rel="noreferrer"
          target="_blank"
        >
          ВКонтакте
        </a>
      </span>
    </Title>
    <Gallery />
  </WithStyles(ForwardRef(Container))>
</styled.section>
`;
